src/samples/coap/lwm2m-client.c

```diff
@@ -322,30 +529,185 @@ read_server_obj(void *instance_data, void *user_data,
+            r = sol_lwm2m_tlv_get_int(tlv, &instance_ctx->lifetime);
+            break;
+        case SERVER_OBJ_BINDING_RES_ID:
+            r = sol_lwm2m_tlv_get_bytes(tlv, &buf);
+            // TODO: Should I check if exists and unref(instance_ctx->binding) first?
+            instance_ctx->binding = sol_buffer_to_blob(&buf);
+            break;
+        default:
+            // TODO: How to inform the infra of each failed resource id (tlv->id)?
+            fprintf(stderr, "tlv type: %u, ID: %" PRIu16 ", Size: %zu, Content: %.*s"
+                " could not be written to Server Object at /1/%" PRIu16,
+                tlv->type, tlv->id, tlv->content.used,
+                SOL_STR_SLICE_PRINT(sol_buffer_get_slice(&tlv->content)), instance_id);
+            break;
+        }
+    }
+
+    return r;
+}
+
+static int
+create_server_obj(void *user_data, struct sol_lwm2m_client *client,
+    uint16_t instance_id, void **instance_data,
+    enum sol_lwm2m_content_type content_type,
+    const struct sol_str_slice content)
+{
+    struct server_obj_instance_ctx *instance_ctx;
+    struct sol_vector tlvs;
+    int r;
+    uint16_t i;
+    struct sol_lwm2m_tlv *tlv;
+
+    if (content_type != SOL_LWM2M_CONTENT_TYPE_TLV) {
+        fprintf(stderr, "Content type is not in TLV format\n");
+        return -EINVAL;
+    }
+
+    instance_ctx = calloc(1, sizeof(struct server_obj_instance_ctx));
+    if (!instance_ctx) {
+        fprintf(stderr, "Could not alloc memory for server object context\n");
+        return -ENOMEM;
+    }
+
+    r = sol_lwm2m_parse_tlv(content, &tlvs);
+    if (r < 0) {
+        fprintf(stderr, "Could not parse the TLV content\n");
+        goto err_free_tlvs;
+    }
+
+    SOL_VECTOR_FOREACH_IDX (&tlvs, tlv, i) {
+        SOL_BUFFER_DECLARE_STATIC(buf, 64);
+
+        if (tlv->id == SERVER_OBJ_SHORT_RES_ID) {
+            r = sol_lwm2m_tlv_get_int(tlv, &instance_ctx->server_id);
+        } else if (tlv->id == SERVER_OBJ_LIFETIME_RES_ID) {
+            r = sol_lwm2m_tlv_get_int(tlv, &instance_ctx->lifetime);
+        } else if (tlv->id == SERVER_OBJ_BINDING_RES_ID) {
+            r = sol_lwm2m_tlv_get_bytes(tlv, &buf);
+            instance_ctx->binding = sol_buffer_to_blob(&buf);
+        }
+
+        if (r < 0) {
+            fprintf(stderr, "Could not get the tlv value for resource %"
+                PRIu16 "\n", tlv->id);
+            goto err_free_tlvs;
+        }
+    }
+
+    instance_ctx->client = client;
+    *instance_data = instance_ctx;
+    sol_lwm2m_tlv_list_clear(&tlvs);
+    printf("Server object created\n");
+
+    return 0;
+
+err_free_tlvs:
+    sol_lwm2m_tlv_list_clear(&tlvs);
+    free(instance_ctx);
+    return r;
+}
+
+static int
 execute_server_obj(void *instance_data, void *user_data,
     struct sol_lwm2m_client *client, uint16_t instance_id,
     uint16_t res_id, const struct sol_str_slice args)
 {
+    // TODO: should I add this resource to server_obj_instance_ctx?
```

Comment:
Yeah, well... But besides being Executable, it is also "Creatable". So it seems to me that I should be prepared to receive it in the create callback, and thus it should be part of the `instance_data`.

What do you think?